modules/installation-configuring-nutanix-failure-domains.adoc

```diff
@@ -0,0 +1,88 @@
+// Module included in the following assemblies:
+//
+// * installing/installing_nutanix/installing-nutanix-installer-provisioned.adoc
+// * installing/installing_nutanix/installing-restricted-networks-nutanix-installer-provisioned.adoc
+
+:_mod-docs-content-type: PROCEDURE
+[id="installation-configuring-nutanix-failure-domains_{context}"]
+= Configuring failure domains
+
+Failure domains improve the fault tolerance of an {product-title} cluster by distributing control plane and compute machines across multiple Nutanix Prism Elements (clusters).
+
+[TIP]
+====
+It is recommended that three failure domains be configured to ensure high-availability.
+====
+
+.Prerequisites
+
+* You have an installation configuration file (`install-config.yaml`).
+
+.Procedure
+
+. Edit the `install-config.yaml` file and add the following stanza to configure the first failure domain:
++
+[source,yaml]
+----
+# ...
+platform:
+  nutanix:
+    failureDomains:
+    - name: <failure_domain_name>
+      prismElement:
+        name: <prism_element_name>
+        uuid: <prism_element_uuid>
+      subnetUUIDs:
+      - <network_uuid>
+# ...
+----
++
+where:
+
+`<failure_domain_name>`:: Specifies a unique name for the failure domain. The name is limited to 64 or fewer characters, which can include lower-case letters, digits, and a dash (`-`). The dash cannot be in the leading or ending position of the name.
+`<prism_element_name>`:: Optional. Specifies the name of the Prism Element.
+`<prism_element_uuid`>:: Specifies the UUID of the Prism Element.
+`<network_uuid`>:: Specifies the UUID of the Prism Element subnet object. The subnet's IP address prefix (CIDR) should contain the virtual IP addresses that the {product-title} cluster uses. Only one subnet per failure domain (Prism Element) in an {product-title} cluster is supported.
+
+. As required, configure additional failure domains.
+. Do one of the following:
+
+** If compute and control plane machines can share the same set of failure domains, list the failure domain names under the cluster's default machine configuration.
++
+.Example of control plane and compute machines sharing a set of failure domains
++
+[source,yaml]
+----
+# ...
+platform:
+  nutanix:
+    defaultMachinePlatform:
+      failureDomains:
+        - failure-domain-1
+        - failure-domain-2
+        - failure-domain-3
+# ...
+----
+** If compute and control plane machines must use different failure domains, list the failure domain names under the respective machine pools.
++
+.Example of control plane and compute machines using different failure domains
++
+[source,yaml]
+----
+# ...
+controlPlane:
+  nutanix:
+    failureDomains:
+      - failure-domain-1
+      - failure-domain-2
+      - failure-domain-3
+# ...
+compute:
+  nutanix:
+    failureDomains:
+      - failure-domain-1
+      - failure-domain-2
+# ...
+----
```

Comment:
failure domains under the respective machine pools should be under `platform` like
```
controlPlane:
  platform:
    nutanix:
      failureDomains:
        - failure-domain-1
        - failure-domain-2
        - failure-domain-3
compute:
  platform:
    nutanix:
      failureDomains:
        - failure-domain-1
        - failure-domain-2
```

Reply:
Good catch. Done.